atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/service/AsyncTransactionService.java

```diff
@@ -47,5 +48,40 @@ public interface AsyncTransactionService {
      * @return {@link ListenableFuture} containing the map from a transaction start timestamp to transaction commit
      * timestamp, possibly missing entries if relevant transactions have not committed yet
      */
+    @Deprecated
     ListenableFuture<Map<Long, Long>> getAsync(Iterable<Long> startTimestamps);
+
+    /**
+     * Gets the transaction status associated with a given {@code startTimestamp} in a future, potentially computing
+     * it asynchronously. Known future result responses may be cached on the client-side.
+     *
+     * Future result may return unknown transaction status, which means that the transaction in question has been
+     * concluded i.e. its status cannot change.
```

Comment:
What happens if I query the transaction service for some random timestamp that doesn't actually correspond to a transaction?

Reply:
if it is on txn4, it will be considered concluded and hence committed. But for earlier schemas it will be considered in-progress.